File: src/modelos-neumatico/modelos-neumatico.controller.ts
import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
import { ModelosNeumaticoService } from './modelos-neumatico.service';
import { CreateModelosNeumaticoDto } from './dto/create-modelos-neumatico.dto';
import { UpdateModelosNeumaticoDto } from './dto/update-modelos-neumatico.dto';

@Controller('modelos-neumatico')
export class ModelosNeumaticoController {
  constructor(private readonly modelosNeumaticoService: ModelosNeumaticoService) {}

  @Post()
  create(@Body() createModelosNeumaticoDto: CreateModelosNeumaticoDto) {
    return this.modelosNeumaticoService.create(createModelosNeumaticoDto);
  }

  @Get()
  findAll() {
    return this.modelosNeumaticoService.findAll();
  }

  @Get(':id')
  findOne(@Param('id') id: string) {
    return this.modelosNeumaticoService.findOne(+id);
  }

  @Patch(':id')
  update(@Param('id') id: string, @Body() updateModelosNeumaticoDto: UpdateModelosNeumaticoDto) {
    return this.modelosNeumaticoService.update(+id, updateModelosNeumaticoDto);
  }

  @Delete(':id')
  remove(@Param('id') id: string) {
    return this.modelosNeumaticoService.remove(+id);
  }
}

File: src/main/java/com/tallermecanico/controller/ModelosNeumaticoController.java
package com.tallermecanico.controller;

import com.tallermecanico.dto.request.CreateModelosNeumaticoDto;
import com.tallermecanico.dto.request.UpdateModelosNeumaticoDto;
import com.tallermecanico.dto.response.DeleteModelosNeumaticoResponseDto;
import com.tallermecanico.entity.ModelosNeumatico;
import com.tallermecanico.service.ModelosNeumaticoService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/modelos-neumatico")
public class ModelosNeumaticoController {

    private final ModelosNeumaticoService modelosNeumaticoService;

    public ModelosNeumaticoController(ModelosNeumaticoService modelosNeumaticoService) {
        this.modelosNeumaticoService = modelosNeumaticoService;
    }

    @PostMapping
    public ResponseEntity<ModelosNeumatico> create(@Valid @RequestBody CreateModelosNeumaticoDto dto) {
        ModelosNeumatico modelo = modelosNeumaticoService.create(dto);
        return ResponseEntity.status(HttpStatus.CREATED).body(modelo);
    }

    @GetMapping
    public ResponseEntity<List<ModelosNeumatico>> findAll() {
        List<ModelosNeumatico> modelos = modelosNeumaticoService.findAll();
        return ResponseEntity.ok(modelos);
    }

    @GetMapping("/{id}")
    public ResponseEntity<ModelosNeumatico> findById(@PathVariable Integer id) {
        ModelosNeumatico modelo = modelosNeumaticoService.findById(id);
        return ResponseEntity.ok(modelo);
    }

    @PatchMapping("/{id}")
    public ResponseEntity<ModelosNeumatico> update(@PathVariable Integer id,
                                                   @Valid @RequestBody UpdateModelosNeumaticoDto dto) {
        ModelosNeumatico modelo = modelosNeumaticoService.update(id, dto);
        return ResponseEntity.ok(modelo);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<DeleteModelosNeumaticoResponseDto> delete(@PathVariable Integer id) {
        DeleteModelosNeumaticoResponseDto response = modelosNeumaticoService.delete(id);
        return ResponseEntity.ok(response);
    }
}
